google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -52,23 +64,89 @@ def __init__(
         """Retrieves an existing managed dataset given a dataset name or ID.
 
         Args:
-            dataset_name (str):
+            dataset_name: (str)
                 Required. A fully-qualified dataset resource name or dataset ID.
                 Example: "projects/123/locations/us-central1/datasets/456" or
                 "456" when project and location are initialized or passed.
-            project (str):
+            project: (str) = None
                 Optional project to retrieve dataset from. If not set, project
                 set in aiplatform.init will be used.
-            location (str):
+            location: (str) = None
                 Optional location to retrieve dataset from. If not set, location
                 set in aiplatform.init will be used.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials: Optional[auth_credentials.Credentials] = None,
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
+
+        Raises:
+            ValueError if the dataset type of the retrieved dataset resource is
+            not supported by the 'Dataset' class.
         """
 
         super().__init__(project=project, location=location, credentials=credentials)
         self._gca_resource = self._get_gca_resource(resource_name=dataset_name)
+        self._validate_resource_metadata_schema_uri()
+
+    @property
+    def resource_metadata_schema_uri(self) -> str:
+        """The metadata schema uri of this dataset resource."""
+        return self._gca_resource.metadata_schema_uri
+
+    def _validate_resource_metadata_schema_uri(self) -> bool:
+        if self._support_metadata_schema_uris and (
+            self.resource_metadata_schema_uri not in self._support_metadata_schema_uris
+        ):
+            raise ValueError(
+                f"{self.__class__.__name__} class can not be used to retrieve "
+                f"dataset resource {self.resource_name}, check the dataset type"
+            )
+        return True
+
+    @property
+    def metadata_schema_uri(self) -> str:
```

Comment:
Is there a reason we're exposing this? I'm of the opinion that we don't need to expose to as a property if not required.